mapswipe_workers/mapswipe_workers/auth.py

```diff
@@ -47,17 +47,20 @@ def firebaseDB() -> object:
 class postgresDB(object):
     """Helper class for Postgres interactions"""
 
-    _db_connection = None
+    __db_connection = None
     _db_cur = None
 
-    def __init__(self):
-        self._db_connection = psycopg2.connect(
-            database=POSTGRES_DB,
-            host=POSTGRES_HOST,
-            password=POSTGRES_PASSWORD,
-            port=POSTGRES_PORT,
-            user=POSTGRES_USER,
-        )
+    @property
+    def _db_connection(self):
+        if self.__db_connection is None:
```

Comment:
This will setup the connection when first called, as opposed to startup time. I'm wondering if startup time isn't better in the sense that it will raise an error immediately (ie. when the admin is starting the app and paying attention), rather than at some random point later in time, when maybe they're gone and doing something else?
Any particular reason to do it this way?

Reply:
https://github.com/mapswipe/python-mapswipe-workers/pull/562#discussion_r992123210

I think existing uses also create a similar problem as the function is called in each module. Maybe we can discuss it further on this issue.